pyiron_base/project/delayed.py

```diff
@@ -0,0 +1,273 @@
+                    nodes_dict=nodes_dict,
+                    edges_lst=edges_lst,
+                    link_node=node_name,
+                )
+    elif isinstance(obj, dict) and any(
+        [isinstance(v, DelayedObject) for v in obj.values()]
+    ):
+        for k, v in obj.items():
+            nodes_dict, edges_lst = get_graph(
+                obj=v,
+                obj_name=k,
+                nodes_dict=nodes_dict,
+                edges_lst=edges_lst,
+                link_node=node_name,
+            )
+    elif isinstance(obj, list) and any([isinstance(v, DelayedObject) for v in obj]):
+        for k, v in enumerate(obj):
+            nodes_dict, edges_lst = get_graph(
+                obj=v,
+                obj_name=str(k),
+                nodes_dict=nodes_dict,
+                edges_lst=edges_lst,
+                link_node=node_name,
+            )
+    return nodes_dict, edges_lst
+
+
+def recursive_dict_resolve(input_dict: dict) -> dict:
+    """
+    Recursively resolve the dictionary to call result() on all objects of type DelayedObject
+
+    Args:
+        input_dict (dict): dictionary to recursively resolve
+
+    Returns:
+        dict: resolved dictionary
+    """
+    output_dict = {}
+    for k, v in input_dict.items():
+        if isinstance(v, DelayedObject):
+            output_dict[k] = v.result()
+        elif isinstance(v, dict):
+            output_dict[k] = recursive_dict_resolve(input_dict=v)
+        elif isinstance(v, list):
+            output_dict[k] = list(
+                recursive_dict_resolve(
+                    input_dict={k: v for k, v in enumerate(v)}
+                ).values()
+            )
+        else:
+            output_dict[k] = v
+    return output_dict
+
+
+def draw(node_dict: dict, edge_lst: list):
+    """
+    Draw graph of nodes and edges
+
+    Args:
+        node_dict (dict): Dictionary of nodes
+        edge_lst (list): List of edges
+    """
+    import matplotlib.pyplot as plt
+    import networkx as nx
+    from IPython.display import SVG, display
+
+    graph = nx.DiGraph()
+    for k, v in node_dict.items():
+        graph.add_node(k, label=str(k).rsplit("_", 1)[0] + "=" + str(v))
+    for edge in edge_lst:
+        graph.add_edge(edge[1], edge[0])
+    svg = nx.nx_agraph.to_agraph(graph).draw(prog="dot", format="svg")
+    display(SVG(svg))
+    plt.show()
+
+
+class Selector:
+    def __init__(self, obj: object, selector: str):
+        self._obj = obj
+        self._selector = selector
+
+    def __getattr__(self, name: str):
+        if self._selector == "files" and name in self._obj._output_file_lst:
+            obj_copy = self._obj.__copy__()
+            obj_copy._output_file = name
+            return obj_copy
+        elif self._selector == "output" and name in self._obj._output_key_lst:
+            obj_copy = self._obj.__copy__()
+            obj_copy._output_key = name
+            return obj_copy
+        else:
+            raise AttributeError()
```

Comment:
```suggestion
            return self.__getattribute__(name)
```

It's the same thing but at least it shows an error message.